src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs

```diff
@@ -138,14 +138,51 @@ public static IDataView LoadFeaturizedAdultDataset(MLContext mlContext)
                 .Append(mlContext.Transforms.Concatenate("Features", "workclass", "education", "marital-status",
                     "occupation", "relationship", "ethnicity", "native-country", "age", "education-num",
                     "capital-gain", "capital-loss", "hours-per-week"))
-                // Min-max normalized all the features
+                // Min-max normalize all the features
                 .Append(mlContext.Transforms.Normalize("Features"));
 
             var data = reader.Read(dataFile);
             var featurizedData = pipeline.Fit(data).Transform(data);
             return featurizedData;
         }
 
+        public static string DownloadMslrWeb10k()
+        {
+            var fileName = "MSLRWeb10KTrain720kRows.tsv";
+            if (!File.Exists(fileName))
+                Download("https://tlcresources.blob.core.windows.net/datasets/MSLR-WEB10K/MSLR-WEB10K_Fold1.TRAIN.500MB_720k-rows.tsv", fileName);
```

Comment:
>https://tlcresources.blob.core.windows.net/datasets/MSLR-WEB10K/MSLR-WEB10K_Fold1.TRAIN.500MB_720k-rows.tsv [](start = 26, length = 107)

This thing is 500 Mb, why we using it?

Reply:
@Ivanidzo4ka This is the only actual ranking dataset we have. I know we use Adult dataset to do some ranking tests with a fake group id column, but I'm not going to add it to our ranking samples. It will only invite confused questions as to why we are using column X as group id, why are we doing ranking on a binary classification dataset, etc. I think it's best to avoid all of that and show an actual ranking example. 

I can either switch to using the validation set for this example, which is 160MB, or make a smaller subset of the 500MB dataset. What do you think?

Reply:
Can we just generate one?
160 MB is too much.
you generated dataset in which you have 10 features, you have 10 different groups, and label depends on value in feature_x if x is value in group column.
Or since you do example for lightgbm you can use [https://github.com/Microsoft/LightGBM/tree/master/examples/lambdarank](https://github.com/Microsoft/LightGBM/tree/master/examples/lambdarank) ligthgbm dataset

---
In reply to: [260045081](https://github.com/dotnet/machinelearning/pull/2704#discussion_r260045081) [](ancestors = 260045081)

Reply:
Hmm that's not a very easy to follow example. There is no group id in the data, nor does the .conf file specify anything about it. It's very unclear how that is being handled or whether the ranker is being trained with one row per group (and if so, why). I'll just make a small sample of MSLR dataset, something like 10,000 rows, ~7MB.

Reply:
It specifies query file. in which you have list of numbers. take first number, it's gonna be amount of examples for first group. take second number -> number of examples for second group and so on.
same for test.
My main concern was regarding publicity of that MSLR dataset, but it looks like it become public quite a while ago [https://www.microsoft.com/en-us/research/project/mslr/](https://www.microsoft.com/en-us/research/project/mslr/)


---
In reply to: [260058524](https://github.com/dotnet/machinelearning/pull/2704#discussion_r260058524) [](ancestors = 260058524)

Reply:
Yes, and we also have approval to use this in our samples with attribution information in the [README] (https://github.com/dotnet/machinelearning/blob/master/test/data/README.md).